pygeos/strtree.py

```diff
@@ -32,15 +48,41 @@ def __init__(self, geometries, leafsize=5):
     def __len__(self):
         return self._tree.count
 
-    def query(self, envelope):
-        """Return all items whose extent intersect the given envelope.
+    def query(self, geometry, predicate=None):
+        """Return all items whose extent intersect the envelope of the input
+        geometry.  If predicate is provided, these items are limited to those
+        that satisfy the predicate operation when compared against the input
+        geometry.
+
+        If geometry is None, an empty array is returned.
 
         Parameters
         ----------
-        envelope : Geometry
-            The envelope of the geometry is taken automatically.
+        geometry : Geometry
+            The envelope of the geometry is taken automatically for
+            querying the tree.
+        predicate : str, optional (default: None)
+            The predicate to use for testing geometries from the tree
+            that are within the input geometry's envelope.
         """
-        return self._tree.query(envelope)
+
+        if geometry is None:
+            return np.array([], dtype="int")
```

Comment:
This is no longer handled on the C level?

Reply:
And if done here, you can use `dtype=np.intp` for consistency in the dtype

Reply:
I had assumed (but not sure I tested) that passing `None` would be problematic for parsing the function arguments via `PyArg_ParseTuple(args, "O!i", &GeometryType, &geometry, &predicate)`

According to the [docs](https://docs.python.org/3/c-api/arg.html) for `O!`, a `TypeError` would be raised if we pass in something of `NoneType` instead of `GeometryType`, and failing out here returns `NULL` instead of an empty array.

I'm not that familiar with `PyArg_ParseTuple`, so if there is a better way we can handle `None` we could deal with at the C level.

Can certainly change the `dtype` though.

Reply:
Confirmed, passing `None` to the C function as it is now raises
`TypeError: argument 1 must be pygeos.lib.GEOSGeometry, not None`